attribute-masking-service/src/main/java/uk/gov/gchq/palisade/service/attributemask/service/KafkaHealthIndicator.java

```diff
@@ -38,29 +37,33 @@
  * Kafka health indicator. Check that the consumer group can be accessed and is registered with the cluster,
  * if not mark the service as unhealthy.
  */
-@Component
+@Component("kafka")
 @ConditionalOnEnabledHealthIndicator("kafka")
 public class KafkaHealthIndicator implements HealthIndicator {
+
     private static final Logger LOGGER = LoggerFactory.getLogger(KafkaHealthIndicator.class);
+
+    private final String groupId;
     private final AdminClient adminClient;
-    @Value("${akka.kafka.consumer.kafka-clients.group.id}")
-    private String groupId;
 
     /**
      * Requires the AdminClient to interact with Kafka
      *
+     * @param groupId     of the cluster
      * @param adminClient of the cluster
      */
-    public KafkaHealthIndicator(final AdminClient adminClient) {
+    public KafkaHealthIndicator(@Value("${akka.kafka.consumer.kafka-clients.group.id}") final String groupId, final AdminClient adminClient) {
+        this.groupId = groupId;
         this.adminClient = adminClient;
     }
 
     @Override
     public Health getHealth(final boolean includeDetails) {
-        return Optional.of(performCheck())
-                .filter(healthy -> healthy)
-                .map(up -> Health.up().withDetail("group", this.groupId).build())
-                .orElseGet(() -> Health.down().withDetail("group", this.groupId).build());
+        if (performCheck()) {
+            return Health.up().withDetail("group", this.groupId).build();
+        } else {
+            return Health.down().withDetail("group", this.groupId).build();
+        }
     }
```

Comment:
Don't override this, spring will provide a better default (i.e. actually using the includeDetails flag).
See `org.springframework.boot.actuate.health.HealthIndicator` class:
```
public interface HealthIndicator extends HealthContributor {

	/**
	 * Return an indication of health.
	 * @param includeDetails if details should be included or removed
	 * @return the health
	 * @since 2.2.0
	 */
	default Health getHealth(boolean includeDetails) {
		Health health = health();
		return includeDetails ? health : health.withoutDetails();
	}

	/**
	 * Return an indication of health.
	 * @return the health
	 */
	Health health();

}
```
You should only implement the health() method